src/backend/renderer/utils.rs

```diff
@@ -19,31 +18,72 @@ use wayland_server::protocol::{wl_buffer::WlBuffer, wl_surface::WlSurface};
 
 #[derive(Default)]
 pub(crate) struct SurfaceState {
+    pub(crate) commit_count: usize,
```

Comment:
`commit_count` may be a bit misleading as it is only incremented if there was a `BufferAssignment`

Reply:
Well true, but it essentially is, what we are seeing as a commit, because merged commits are indistinguishable due to the merging. Also `on_commit_buffer_handler` is meant to be called on each commit anywhere, any usage other than that is undefined behavior anyway.